README.md
# TripMobileExpo

This app will be used on android and hopefully IOS!


CHANGELOG.md
All notable changes to this project will be documented in this file.

### 0.2.0

├─ Initial beta release

### 0.2.1

├─ Updated to react-native@0.68.2
├─ Installed react-native-dropdown-picker@5.4.2
├─ Installed react-native-svg@12.3.0
├─ Installed react-native-svg-transformer@1.0.0
├─ Updated to expo-updates@0.13.2
├─ Updated to expo-web-browser@10.2.1
├─ Updated to react@17.0.2
├─ Added SVG support
├─ Replaced home page icons with SVG icons


modules/test.js
import * as React from 'react';
import {
  SafeAreaView,
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
} from "react-native"

import SearchableDropdown from 'react-native-searchable-dropdown';

// For each dictionary in the allDrugData list, find the "name" key and add it to a list of index-title pairs
const drugNames = Object.keys(allDrugData).map((key, index) => {
  // console.log(allDrugData[key]["name"] + allDrugData[key]["url"])
  // console.log(allDrugData[key]["summary"])
  return {
    id: index,
    name: allDrugData[key]["name"]
  }
})


// Okay this is stupid, but I'm not sure how to do it better
// We pop out the last element of the array, because the above MAP function adds an 'undefined' element to the array
// The above function adds an undefined element because that's how the JSON file is imported, for some reason
// TODO - JSON UNDEFINED ISSUE
drugNames.pop()

// console.log(drugNames)
var drug_a = "";
var drug_b = "";

function TestScreen() {
  // Define objects to hold search data
  const [drugA, setDrugA] = React.useState('');
  const [drugB, setDrugB] = React.useState('');
  // Much like Factsheets, the results only display after you search
  const [shouldShow, setShouldShow] = React.useState(false);
  // Define object to store results
  const [results, setResults] = React.useState({});


  function getDrugCombo() {
    // console.log("started combo!")
    // console.log("drug_a is: " + drug_a)
    // console.log("drug_b is: " + drug_b)
    var comboData = ""
    if (drug_a != "" && drug_b != "") {
      // console.log("both drugs are set!")
      for (var i = 0; i < Object.keys(allDrugData).length; i++) {
        // TODO - JSON UNDEFINED ISSUE - Same as above
        if (!!allDrugData[i]) {
          // console.debug("allDrugData[i]['name'] is: " + allDrugData[i]['name'])
          if (allDrugData[i]['name'] == drug_a) {
            // console.debug("found drug_a: " + drug_a)
            drugData = allDrugData[i]
            if (!!allDrugData[i]['interactions']) {
              // console.debug("drug_a has interactions!")
              // console.debug("drugData[i]['interactions'] is: " + allDrugData[i]['interactions'])
              for (var j = 0; j < allDrugData[i]['interactions'].length; j++) {
                // console.debug("j is: " + j)
                // console.debug("allDrugData[i]['interactions'][j]['name'] is: " + allDrugData[i]['interactions'][j]['name'])
                if (allDrugData[i]['interactions'][j]['name'] == drug_b) {
                  // console.debug("found drug_b: " + drug_b)
                  comboData = allDrugData[i]['interactions'][j]['status']
                  // console.debug("comboData is: " + comboData)
                  setResults(comboData)
                  setShouldShow(true)
                  break
                }
              }
              if (comboData == "") {
                setResults("Interaction data not found between " + drug_a + " and " + drug_b)
                setShouldShow(true)
                break
              }
            } else {
              setResults("Interaction data not found for " + drug_a)
              setShouldShow(true)
              break
            }
          }
        }
      }
    }
  }

  return (
    <SafeAreaView style={[{ flex: 1 }]}>
      <View style={styles.container}>
        <View style={styles.row}>
        <View>
            <SearchableDropdown
              onItemSelect={(item) => {
                // console.log(item.name)
                drug_a = item.name
                setDrugA(item.name)
                getDrugCombo()
              }}
              containerStyle={{ padding: 5 }}
              itemStyle={{
                padding: 10,
                marginTop: 2,
                backgroundColor: '#ddd',
                borderColor: '#bbb',
                borderWidth: 1,
                borderRadius: 5,
              }}
              itemTextStyle={{ color: '#222' }}
              itemsContainerStyle={{ maxHeight: 140 }}
              items={drugNames}
              resetValue={false}
              textInputProps={
                {
                  placeholder: "Search substance here!",
                  underlineColorAndroid: "transparent",
                  style: {
                      padding: 12,
                      borderWidth: 1,
                      borderColor: '#ccc',
                      borderRadius: 5,
                      backgroundColor: '#ffffff'
                  },
                }
              }
              listProps={
                {
                  nestedScrollEnabled: true,
                }
              }
            />
          </View>
          <View>
            <SearchableDropdown
              onItemSelect={(item) => {
                // console.log(item.name)
                drug_b = item.name
                setDrugB(item.name)
                getDrugCombo()
              }}
              containerStyle={{ padding: 5 }}
              itemStyle={{
                padding: 10,
                marginTop: 2,
                backgroundColor: '#ddd',
                borderColor: '#bbb',
                borderWidth: 1,
                borderRadius: 5,
              }}
              itemTextStyle={{ color: '#222' }}
              itemsContainerStyle={{ maxHeight: 140 }}
              items={drugNames}
              resetValue={false}
              textInputProps={
                {
                  placeholder: "Search substance here!",
                  underlineColorAndroid: "transparent",
                  style: {
                      padding: 12,
                      borderWidth: 1,
                      borderColor: '#ccc',
                      borderRadius: 5,
                      backgroundColor: '#ffffff'
                  },
                }
              }
              listProps={
                {
                  nestedScrollEnabled: true,
                }
              }
            />
          </View>
        </View>
        <View style={styles.row}>
          <TouchableOpacity
            style={styles.button}
          >
            <Text style={styles.buttonText}>{drugA}</Text>
          </TouchableOpacity>
          <TouchableOpacity
            style={styles.button}
          >
            <Text style={styles.buttonText}>{drugB}</Text>
          </TouchableOpacity>
        </View>
        {shouldShow ? (
          <View style={styles.container}>
            <Text style={styles.buttonText}>{results}</Text>
          </View>
        ) : null}
      </View>
    </SafeAreaView>
  )
}

const color_dict = {
  black: '#000000',
  red: '#FF0000',
  green: '#00FF00',
  blue: '#0000FF',
  white: '#FFFFFF',
  purple: '#4B0082',
};

const styles = StyleSheet.create({
  flex: {
    flex: 1
  },
  justifyCenter: {
    justifyContent: 'center'
  },
  row: {
    flexDirection: 'row',
  },
  input: {
    height: 50,
    margin: 12,
    borderWidth: 1,
    padding: 10,
    backgroundColor: color_dict.white,
  },
  container: {
    flex: 1,
    backgroundColor: color_dict.black,
    alignItems: 'center',
    // padding: 20,
  },
  logo: {
    width: 305,
    height: 159,
    marginBottom: 10,
  },
  button: {
    width: '50%',
    padding: 20,
    borderRadius: 5,
  },
  searchButton: {
    backgroundColor: color_dict.purple,
    padding: 20,
    borderRadius: 5,
    height: 10,
    justifyContent: 'space-around'
  },
  doubleButton: {
    width: '100%',
    padding: 20,
    borderRadius: 5,
  },
  buttonText: {
    fontSize: 20,
    color: color_dict.white,
    alignSelf: 'center',
  },
  icon: {
    height: 60,
    width: 60,
    alignSelf: 'center',
  },
  text: {
    fontSize: 16,
    color: color_dict.white,
  },
  header: {
    backgroundColor: color_dict.purple,
    padding: 10,
  },
  aboutHeaderText: {
    fontSize: 20,
    color: color_dict.white,
    alignSelf: 'center'
  },
  headerText: {
    textAlign: 'center',
    fontSize: 16,
    fontWeight: '500',
    color: color_dict.white,
  },
  content: {
    padding: 20,
  },
  active: {
    backgroundColor: color_dict.black,
  },
  inactive: {
    backgroundColor: color_dict.black,
  },
  scrollContainer: {
    flex: 1
  },
  title: {
    textAlign: "center",
    fontSize: 25,
    marginBottom: 50
  },
  sectionTitle: {
    fontWeight: "bold",
    marginBottom: 3,
  },
  section: {
    marginBottom: 40,
  },
});

module.exports = { TestScreen };


App.js
import * as React from 'react';
// React elements to use
import { StyleSheet } from 'react-native';
// Both below are for navigation
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';

const about = require("./modules/about");
// const test = require("./modules/test");
const settings = require("./modules/settings");
const contact = require("./modules/contact");
const combo = require("./modules/combo");
const facts = require("./modules/facts");
const home = require("./modules/home");

function App() {
  return (
    <NavigationContainer>
      <Stack.Navigator
        // These settings control the navigation bar
        // By default this bar does not appear on the homepage
        screenOptions={{
          headerStyle: {
            backgroundColor: '#000',
          },
          headerTintColor: '#fff',
          headerTitleStyle: {
            fontWeight: 'bold',
          },
          headerTitleAlign: 'center',
        }}
      >
        {/* Every screen the user will see must be defined here first */}
        {/* Then use the navigation module and touchableOpacity  */}
        <Stack.Screen name="Home" component={home.HomeScreen} options={{ headerShown: false }} />
        <Stack.Screen name="TripSit's Factsheet Search" component={facts.FactsScreen} />
        <Stack.Screen name="TripSit's Combo Search" component={combo.ComboScreen} />
        <Stack.Screen name="Contact" component={contact.ContactScreen} />
        <Stack.Screen name="Settings" component={settings.SettingsScreen} />
        <Stack.Screen name="About" component={about.AboutScreen} />
        {/* <Stack.Screen name="Test" component={test.TestScreen} /> */}
      </Stack.Navigator>
    </NavigationContainer>
  );
}

const color_dict = {
  black: '#000000',
  red: '#FF0000',
  green: '#00FF00',
  blue: '#0000FF',
  white: '#FFFFFF',
  purple: '#4B0082',
};

const styles = StyleSheet.create({
  flex: {
    flex: 1
  },
  justifyCenter: {
    justifyContent: 'center'
  },
  row: {
    flexDirection: 'row',
  },
  input: {
    height: 50,
    margin: 12,
    borderWidth: 1,
    padding: 10,
    backgroundColor: color_dict.white,
  },
  container: {
    flex: 1,
    backgroundColor: color_dict.black,
    alignItems: 'center',
  },
  logo: {
    width: 305,
    height: 159,
    marginBottom: 10,
  },
  button: {
    width: '50%',
    padding: 20,
    borderRadius: 5,
  },
  searchButton: {
    backgroundColor: color_dict.purple,
    padding: 20,
    borderRadius: 5,
    height: 10,
    justifyContent: 'space-around'
  },
  doubleButton: {
    width: '100%',
    padding: 20,
    borderRadius: 5,
  },
  buttonText: {
    fontSize: 20,
    color: color_dict.white,
    alignSelf: 'center',
  },
  icon: {
    height: 60,
    width: 60,
    alignSelf: 'center',
  },
  text: {
    fontSize: 16,
    color: color_dict.white,
  },
  header: {
    backgroundColor: color_dict.purple,
    padding: 10,
  },
  aboutHeaderText: {
    fontSize: 20,
    color: color_dict.white,
    alignSelf: 'center'
  },
  headerText: {
    textAlign: 'center',
    fontSize: 16,
    fontWeight: '500',
    color: color_dict.white,
  },
  content: {
    padding: 20,
  },
  active: {
    backgroundColor: color_dict.black,
  },
  inactive: {
    backgroundColor: color_dict.black,
  },
});

const Stack = createNativeStackNavigator();

export default App;

.eslintrc.js
module.exports = {
  env: {
    browser: true,
    // es2021: true,
    node: true,
    jest: true,
    'react-native/react-native': true,
  },
  extends: [
    'plugin:react/recommended',
    'airbnb',
  ],
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
    ecmaVersion: 13,
    sourceType: 'module',
  },
  plugins: [
    'react',
    'react-native',
    'react-hooks',
  ],
  rules: {
    'no-use-before-define': 'off',
    'react/jsx-filename-extension': 'off',
    'react/prop-types': 'off',
    'comma-dangle': 'off',
    'padded-blocks': 'off',
    'arrow-body-style': 'off',
    'react-hooks/exhaustive-deps': 'warn',
    'react-native/no-unused-styles': 2,
    'react-native/split-platform-components': 2,
    'react-native/no-inline-styles': 2,
    'react-native/no-color-literals': 2,
    'react-native/no-raw-text': 2,
    'react-native/no-single-element-style-arrays': 2,
    'no-underscore-dangle': 'off',
  },
  globals: {
    fetch: false
  }
};


modules/home.js
import * as React from "react";
// React elements to use
import { Image, StyleSheet, Text, View, TouchableOpacity } from "react-native";
// This controls the bar at the top of the phone that diplays time, battery, etc
import { StatusBar } from "expo-status-bar";
// For opening the web browser on links
import * as WebBrowser from "expo-web-browser";

function handleOpenWithWebBrowser(url) {
  // This allows opening of links in the web browser
  // It's at the top level so any page can use this function
  WebBrowser.openBrowserAsync(url);
}

// Images
import logo from "../assets/images/logo.jpg";
import ChatImage from "../assets/svg/trippy_chat.svg";
import FactsheetsImage from "../assets/svg/trippy_document.svg";
import CombinationsImage from "../assets/svg/trippy_flasks.svg";
import WikiImage from "../assets/svg/trippy_globe.svg";
// import settingsImage from '../assets/images/gears.png';
import ContactImage from "../assets/svg/trippy_contact.svg";
import AboutImage from "../assets/svg/trippy_about.svg";

function HomeScreen({ navigation }) {
  // Set the display of the status bar to false
  const [displayStatusBar] = React.useState(false);
  return (
    <View style={styles.container}>
      {/* StatusBar is hidden on this page, which hides it on all other pages */}
      <StatusBar hidden={displayStatusBar} />
      {/* <TouchableOpacity onPress={() => {navigation.navigate('Test');}}> */}
      <TouchableOpacity>
        <Image source={logo} style={styles.logo} />
      </TouchableOpacity>
      <View>
        <View style={styles.row}></View>

        <View style={styles.row}>
          <TouchableOpacity
            onPress={() => navigation.navigate("TripSit's Factsheet Search")}
            style={styles.button}
          >
            <FactsheetsImage width={100} height={100} style={styles.icon} />
            <Text style={styles.buttonText}>Facts</Text>
          </TouchableOpacity>
          <TouchableOpacity
            onPress={() => navigation.navigate("TripSit's Combo Search")}
            style={styles.button}
          >
            <CombinationsImage width={100} height={100} style={styles.icon} />
            <Text style={styles.buttonText}>Combos</Text>
          </TouchableOpacity>
        </View>

        <View style={styles.row}>
          <TouchableOpacity
            onPress={() => {
              handleOpenWithWebBrowser("https://wiki.tripsit.me");
            }}
            style={styles.button}
          >
            <WikiImage width={100} height={100} style={styles.icon} />
            <Text style={styles.buttonText}>Wiki</Text>
          </TouchableOpacity>
          <TouchableOpacity
            onPress={() => {
              handleOpenWithWebBrowser("https://chat.tripsit.me");
            }}
            style={styles.button}
          >
            <ChatImage width={100} height={100} style={styles.icon} />
            <Text style={styles.buttonText}>Chat</Text>
          </TouchableOpacity>
        </View>

        <View style={styles.row}>
          <TouchableOpacity
            onPress={() => {
              navigation.navigate("Contact");
            }}
            style={styles.button}
          >
            <ContactImage width={100} height={100} style={styles.icon} />
            <Text style={styles.buttonText}>Contact</Text>
          </TouchableOpacity>
          <TouchableOpacity onPress={() => navigation.navigate("About")} style={styles.button}>
            <AboutImage width={100} height={100} style={styles.icon} />
            <Text style={styles.buttonText}>About</Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
}

const color_dict = {
  black: "#000000",
  red: "#FF0000",
  green: "#00FF00",
  blue: "#0000FF",
  white: "#FFFFFF",
  purple: "#4B0082",
};

const styles = StyleSheet.create({
  flex: {
    flex: 1,
  },
  justifyCenter: {
    justifyContent: "center",
  },
  row: {
    flexDirection: "row",
    textAlign: "center",
    border: "1px solid white",
  },
  input: {
    height: 50,
    margin: 12,
    borderWidth: 1,
    padding: 10,
    backgroundColor: color_dict.white,
  },
  container: {
    flex: 1,
    backgroundColor: color_dict.black,
    alignItems: "center",
  },
  logo: {
    width: 325,
    height: 159,
    marginBottom: 10,
  },
  button: {
    width: "50%",
    padding: 20,
    borderRadius: 5,
  },
  searchButton: {
    backgroundColor: color_dict.purple,
    padding: 20,
    borderRadius: 5,
    height: 10,
    justifyContent: "space-around",
  },
  doubleButton: {
    width: "100%",
    padding: 20,
    borderRadius: 5,
  },
  buttonText: {
    fontSize: 20,
    color: color_dict.white,
    alignSelf: "center",
  },
  icon: {
    alignSelf: "center",
  },
  text: {
    fontSize: 16,
    color: color_dict.white,
  },
  header: {
    backgroundColor: color_dict.purple,
    padding: 10,
  },
  aboutHeaderText: {
    fontSize: 20,
    color: color_dict.white,
    alignSelf: "center",
  },
  headerText: {
    textAlign: "center",
    fontSize: 16,
    fontWeight: "500",
    color: color_dict.white,
  },
  content: {
    padding: 20,
  },
  active: {
    backgroundColor: color_dict.black,
  },
  inactive: {
    backgroundColor: color_dict.black,
  },
});

module.exports = { HomeScreen };


modules/about.js
import * as React from "react";
// React elements to use
import { ScrollView, Image, StyleSheet, Text, View, TouchableOpacity } from "react-native";
// For opening the web browser on links
import * as WebBrowser from "expo-web-browser";

const color_dict = {
  black: "#000000",
  red: "#FF0000",
  green: "#00FF00",
  blue: "#0000FF",
  white: "#FFFFFF",
  purple: "#4B0082",
};

const styles = StyleSheet.create({
  flex: {
    flex: 1,
  },
  justifyCenter: {
    justifyContent: "center",
  },
  row: {
    flexDirection: "row",
  },
  input: {
    height: 50,
    margin: 12,
    borderWidth: 1,
    padding: 10,
    backgroundColor: color_dict.white,
  },
  container: {
    flex: 1,
    backgroundColor: color_dict.black,
    alignItems: "center",
    padding: 25,
  },
  logo: {
    width: 305,
    height: 159,
    marginBottom: 10,
  },
  button: {
    width: "50%",
    padding: 20,
    borderRadius: 5,
  },
  searchButton: {
    backgroundColor: color_dict.purple,
    padding: 20,
    borderRadius: 5,
    height: 10,
    justifyContent: "space-around",
  },
  doubleButton: {
    width: "100%",
    padding: 20,
    borderRadius: 5,
  },
  buttonText: {
    fontSize: 20,
    color: color_dict.white,
    alignSelf: "center",
  },
  icon: {
    height: 60,
    width: 60,
    alignSelf: "center",
  },
  text: {
    fontSize: 16,
    color: color_dict.white,
  },
  header: {
    backgroundColor: color_dict.purple,
    padding: 10,
  },
  aboutHeaderText: {
    fontSize: 20,
    color: color_dict.white,
    alignSelf: "center",
  },
  headerText: {
    textAlign: "center",
    fontSize: 16,
    fontWeight: "500",
    color: color_dict.white,
  },
  content: {
    padding: 20,
  },
  active: {
    backgroundColor: color_dict.black,
  },
  inactive: {
    backgroundColor: color_dict.black,
  },
  topSpacer: {
    marginTop: 15,
  },
  rightSpacer: {
    marginRight: 15,
  },
});

function handleOpenWithWebBrowser(url) {
  // This allows opening of links in the web browser
  // It's at the top level so any page can use this function
  WebBrowser.openBrowserAsync(url);
}

import facebookImage from "../assets/images/facebook.png";
import redditImage from "../assets/images/reddit.png";
import twitterImage from "../assets/images/twitter.png";
import bitcointImage from "../assets/images/bitcoin.png";

function AboutScreen() {
  const about =
    "This app is created by TripSit, an organisation which helps to provide factual information about drugs and how to reduce the harms involved in using them. We also have an active IRC (internet relay chat) network where we provide tripsitting services, harm reduction advice, and general chat.";
  const disclaimer =
    "Although we have a team dedicated to keeping the information on this app up to date, it is not always possible to provide entirely accurate information on the safety level of drugs. The information here should be used as guidelines only, and it is important to do your own research from multiple sources before ingesting a substance. We also strongly advise using a testing kit and scales to ensure you are taking the correct dosage. These can both be bought online for reasonable prices.";
  const support =
    "TripSit is a completely free service run by volunteers. If you wish to help out, feel free to join the IRC, follow and share our content on social media, or make a donation to keep the servers running.";
  return (
    <View style={styles.container}>
      <ScrollView>
        <Text style={styles.aboutHeaderText}>About TripSit</Text>
        <Text style={styles.text}>{about}</Text>
        <Text style={[styles.aboutHeaderText, styles.topSpacer]}>Disclaimer</Text>
        <Text style={styles.text}>{disclaimer}</Text>
        <Text style={[styles.aboutHeaderText, styles.topSpacer]}>Support TripSit</Text>
        <Text style={styles.text}>{support}</Text>
        <View style={[styles.row, styles.justifyCenter, styles.topSpacer]} contentD>
          <TouchableOpacity
            onPress={() => {
              handleOpenWithWebBrowser("https://www.facebook.com/groups/tripsit/");
            }}
            style={styles.rightSpacer}
          >
            <Image source={facebookImage} style={styles.icon} />
          </TouchableOpacity>
          <TouchableOpacity
            style={styles.rightSpacer}
            onPress={() => {
              handleOpenWithWebBrowser("https://www.reddit.com/r/TripSit/");
            }}
          >
            <Image source={redditImage} style={styles.icon} />
          </TouchableOpacity>
          <TouchableOpacity
            onPress={() => {
              handleOpenWithWebBrowser("https://twitter.com/teamtripsit");
            }}
            style={styles.rightSpacer}
          >
            <Image source={twitterImage} style={styles.icon} />
          </TouchableOpacity>
          <TouchableOpacity
            onPress={() => {
              handleOpenWithWebBrowser("https://tripsit.me/donate/");
            }}
          >
            <Image source={bitcointImage} style={styles.icon} />
          </TouchableOpacity>
        </View>
      </ScrollView>
    </View>
  );
}

module.exports = { AboutScreen };


modules/combo.js
import * as React from "react";
import { SafeAreaView, StyleSheet, Text, View, TouchableOpacity } from "react-native";
import * as allDrugData from "../assets/data/drug_db_combined.json";
import * as combo_definitions from "../assets/data/combo_definitions.json";
import SearchableDropdown from "react-native-searchable-dropdown";
// const logger = require('./utils/logger.js');
// const PREFIX = require('path').parse(__filename).name;
const PREFIX = "combo";

// For each dictionary in the allDrugData list, find the "name" key and add it to a list of index-title pairs
const drugNames = Object.keys(allDrugData).map((key, index) => {
  // console.log(allDrugData[key]["name"] + allDrugData[key]["url"])
  // console.log(allDrugData[key]["summary"])
  return {
    id: index,
    name: allDrugData[key]["name"],
  };
});

// Okay this is stupid, but I'm not sure how to do it better
// We pop out the last element of the array, because the above MAP function adds an 'undefined' element to the array
// The above function adds an undefined element because that's how the JSON file is imported, for some reason
// TODO - JSON UNDEFINED ISSUE
drugNames.pop();

// console.log(drugNames)
var drug_a = "";
var drug_b = "";

function ComboScreen() {
  // Define objects to hold search data
  const [drugA, setDrugA] = React.useState("");
  const [drugB, setDrugB] = React.useState("");
  // Much like Factsheets, the results only display after you search
  const [shouldShow, setShouldShow] = React.useState(false);
  // Define object to store results
  const [results, setResults] = React.useState({});

  function getDrugCombo() {
    // console.log("started combo!")
    // console.log("drug_a is: " + drug_a)
    // console.log("drug_b is: " + drug_b)
    let comboData = false;
    if (drug_a != "" && drug_b != "") {
      // console.log("both drugs are set!");

      console.log("combo_definitions :: ", combo_definitions);
      console.log("allDrugData :: ", allDrugData);
      console.log("drug_a :: ", drug_a);
      console.log("drug_b :: ", drug_b);
      for (var i = 0; i < Object.keys(allDrugData).length; i++) {
        // TODO - JSON UNDEFINED ISSUE - Same as above
        if (!!allDrugData[i]) {
          // console.debug("allDrugData[i]['name'] is: " + allDrugData[i]['name'])
          if (allDrugData[i]["name"] == drug_a) {
            // console.debug("found drug_a: " + drug_a);
            //drugData = allDrugData[i];
            if (!!allDrugData[i]["interactions"]) {
              // console.debug(`${drug_a} interactions: ${allDrugData[i]["interactions"].length}`);
              for (var j = 0; j < allDrugData[i]["interactions"].length; j++) {
                // console.debug(`Interaction ${j} is: ${allDrugData[i]["interactions"][j]["name"]}`);
                if (allDrugData[i]["interactions"][j]["name"] == drug_b) {
                  const status = allDrugData[i]["interactions"][j]["status"];
                  // console.debug(`Found interaction with ${drug_b}: ${status}`);
                  // Loop through combo_definitions and find the object where "status" is equal to result
                  for (var k = 0; k < Object.keys(combo_definitions).length; k++) {
                    if (combo_definitions[k]["status"] == status) {
                      // console.debug(`Found combo_defs: ${combo_definitions[k]["status"]}`);
                      const definition = combo_definitions[k]["definition"];
                      const emoji = combo_definitions[k]["emoji"];
                      const svg = combo_definitions[k]["svg"];
                      const color = combo_definitions[k]["color"];
                      const title = `${emoji} ${status} ${emoji}`;

                      comboData = true;

                      setResults({
                        title: title,
                        definition: definition,
                        color: color,
                        color_code: color_dict[color.toLowerCase()],
                      });
                      setShouldShow(true);
                      break;
                    }
                  }
                  break;
                }
              }

              if (!comboData) {
                setResults({
                  error: `Interaction data not found between ${drug_a} and ${drug_b}`,
                });
                setShouldShow(true);
                break;
              }
            } else {
              setResults({
                error: `Interaction data not found for ${drug_a}`,
              });
              setShouldShow(true);
              break;
            }
          }
        }
      }
    }
  }

  const comboTitleStyle = function (color_code) {
    return {
      fontSize: 20,
      color: color_dict.white,
      alignSelf: "center",
      backgroundColor: color_code,
      padding: 5,
    };
  };

  return (
    <SafeAreaView style={[{ flex: 1 }]}>
      <View style={styles.container}>
        <View style={styles.row}>
          <View>
            <SearchableDropdown
              onItemSelect={(item) => {
                // console.log(item.name)
                drug_a = item.name;
                setDrugA(item.name);
                getDrugCombo();
              }}
              containerStyle={{ padding: 5 }}
              itemStyle={{
                padding: 10,
                marginTop: 2,
                backgroundColor: "#ddd",
                borderColor: "#bbb",
                borderWidth: 1,
                borderRadius: 5,
              }}
              itemTextStyle={{ color: "#222" }}
              itemsContainerStyle={{ maxHeight: 140 }}
              items={drugNames}
              resetValue={false}
              textInputProps={{
                placeholder: "Search substance here!",
                underlineColorAndroid: "transparent",
                style: {
                  padding: 12,
                  borderWidth: 1,
                  borderColor: "#ccc",
                  borderRadius: 5,
                  backgroundColor: "#ffffff",
                },
              }}
              listProps={{
                nestedScrollEnabled: true,
              }}
            />
          </View>
          <View>
            <SearchableDropdown
              onItemSelect={(item) => {
                // console.log(item.name)
                drug_b = item.name;
                setDrugB(item.name);
                getDrugCombo();
              }}
              containerStyle={{ padding: 5 }}
              itemStyle={{
                padding: 10,
                marginTop: 2,
                backgroundColor: "#ddd",
                borderColor: "#bbb",
                borderWidth: 1,
                borderRadius: 5,
              }}
              itemTextStyle={{ color: "#222" }}
              itemsContainerStyle={{ maxHeight: 140 }}
              items={drugNames}
              resetValue={false}
              textInputProps={{
                placeholder: "Search substance here!",
                underlineColorAndroid: "transparent",
                style: {
                  padding: 12,
                  borderWidth: 1,
                  borderColor: "#ccc",
                  borderRadius: 5,
                  backgroundColor: "#ffffff",
                },
              }}
              listProps={{
                nestedScrollEnabled: true,
              }}
            />
          </View>
        </View>
        <View style={styles.row}>
          <TouchableOpacity style={styles.button}>
            <Text style={styles.buttonText}>{drugA}</Text>
          </TouchableOpacity>
          <TouchableOpacity style={styles.button}>
            <Text style={styles.buttonText}>{drugB}</Text>
          </TouchableOpacity>
        </View>
        {console.log("shouldShow :: ", shouldShow)}
        {console.log("results :: ", results)}
        <View style={styles.container}>
          {!results.error ? (
            <>
              <Text style={comboTitleStyle(results.color_code)}>{results.title}</Text>
              <Text style={styles.comboDefineStyle}>{results.definition}</Text>
            </>
          ) : (
            <Text style={styles.comboDefineStyle}>{results.error}</Text>
          )}
        </View>
      </View>
    </SafeAreaView>
  );
}

const color_dict = {
  red: "#FF0000",
  orange: "#D98427",
  //yellow: "#D5C625",
  green: "#00FF00",
  aqua: "#006CB3",
  blue: "#35AFFF",
  dark_blue: "#0A89DD",
  purple: "#4B0082",
  black: "#000000",
  white: "#FFFFFF",
};

const styles = StyleSheet.create({
  flex: {
    flex: 1,
  },
  justifyCenter: {
    justifyContent: "center",
  },
  row: {
    flexDirection: "row",
  },
  input: {
    height: 50,
    margin: 12,
    borderWidth: 1,
    padding: 10,
    backgroundColor: color_dict.white,
  },
  container: {
    flex: 1,
    backgroundColor: color_dict.black,
    alignItems: "center",
    // padding: 20,
  },
  logo: {
    width: 305,
    height: 159,
    marginBottom: 10,
  },
  button: {
    width: "50%",
    padding: 20,
    borderRadius: 5,
  },
  searchButton: {
    backgroundColor: color_dict.purple,
    padding: 20,
    borderRadius: 5,
    height: 10,
    justifyContent: "space-around",
  },
  doubleButton: {
    width: "100%",
    padding: 20,
    borderRadius: 5,
  },
  buttonText: {
    fontSize: 20,
    color: color_dict.white,
    alignSelf: "center",
  },
  icon: {
    height: 60,
    width: 60,
    alignSelf: "center",
  },
  text: {
    fontSize: 16,
    color: color_dict.white,
  },
  header: {
    backgroundColor: color_dict.purple,
    padding: 10,
  },
  aboutHeaderText: {
    fontSize: 20,
    color: color_dict.white,
    alignSelf: "center",
  },
  headerText: {
    textAlign: "center",
    fontSize: 16,
    fontWeight: "500",
    color: color_dict.white,
  },
  content: {
    padding: 20,
  },
  active: {
    backgroundColor: color_dict.black,
  },
  inactive: {
    backgroundColor: color_dict.black,
  },
  scrollContainer: {
    flex: 1,
  },
  title: {
    textAlign: "center",
    fontSize: 25,
    marginBottom: 50,
  },
  sectionTitle: {
    fontWeight: "bold",
    marginBottom: 3,
  },
  section: {
    marginBottom: 40,
  },
  comboDefineStyle: {
    fontSize: 20,
    color: color_dict.white,
    alignSelf: "center",
    padding: 26,
    paddingTop: 15,
  },
});

module.exports = { ComboScreen };
